enhancements/cluster-logging/elasticsearch-cert-management.md

```diff
@@ -0,0 +1,158 @@
+
+### User Stories
+
+#### As an user of EO, I do not want to have to generate and provide secrets for the Elasticsearch cluster
+
+We want to further simplify using the Elasticsearch Operator and create a single source of truth for the certs the cluster uses.
+
+#### As a consumer of EO, I want a mechanism to have certificates injected into a secret for me to continue mTLS communcation with my Elasticsearch cluster
+
+We want to continue allowing mTLS as this supports the Cluster Logging and Jaeger use cases, however they shouldn't need to do more than notify they need certs injected by EO.
+
+### Implementation Details
+
+#### Assumptions
+
+* The Elasticsearch Operator will implement this in golang (rather than the current script used in CLO)
+
+* The operator will need to be able to store generated certificates for each namespace/unique cluster so that there is no security concerns of clusters cross communicating.
+
+#### Security
+
+Ensuring that signing certificates and configurations are isolated for different clusters is a must for this implementation. It will also need to be able to do this thread safe manner.
+
+### Risks and Mitigations
+
+## Design Details
+
+From a user perspective, there should be no change in any of the CRDs. The only major changes would be no longer generating secrets for EO to use, and needing to annotate a secret (user provided) in the case they want to communicate with ES in a mTLS way.
+
+### secret annotation
+
+We will add an annotation value to check for on secrets in namespaces where an es CR exists:
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: fluentd
+  annotations:
+    secret.openshift.io/elasticsearch-cert-component-name: system.logging.fluentd
+type: Opaque
+data: {}
+```
+
+Where the value will be optional (defaults to the name of the secret) and corresponds to the CN used for the generated certificate.
+
+### Test Plan
+
+#### Unit Testing
+
+* We will need to ensure we have unit tests that verify certificates are generated by the EO
+
+* We will need to ensure we have unit tests that verify secrets are generated by EO with the correct certificates
+
+* We will need to ensure we have unit tests that verify EO generates and injects for annotated secrets
+
+#### Integration and E2E tests
+
+* Given we are moving logic from one operator to another, we need to ensure that there are no regressions with communication between Fluentd, Kibana, IM cronjobs, and Elasticsearch (this should be handled by our smoke test).
+
+### Graduation Criteria
+
+#### GA
+
+* End user documentation
+* Sufficient test coverage
+* Sufficient LTS upgrade strategy
+
+### Version Skew Strategy
+
+We need to have a proper upgrade strategy from our LTS version to the version this is released on (currently 4.6).
+
+## Implementation History
+
+| release|Description|
+|---|---|
+|5.2| **GA** - Initial release
+
+## Drawbacks
+
+None
+
+## Alternatives
+
+None
```

Comment:
1. Leave the contract between the CLO and EO and Jaeger and EO the way it is and document it more clearly.
2. Take a step further and move certificate generation into an OCP-cluster-wide CA, as there's nothing specific to EO, CLO or Jaeger in this code.

Reply:
true, I should be clear that the alternative is to leave it as it currently stands... "None" is misleading